### PR TITLE
Add user agent tag for httpstats package

### PR DESCRIPTION
**Pull request by systemizer**
Curious whether you've thought or would be willing to add user agent as a tag for http stats. It's unbounded in the number of permutations so it could result in high cardinality. At the same time, for internal services, where user agents are generally constrained, it would be very helpful. Could also update the PR to add configuration such that user agent is by default disabled, but give the user the ability to enable it for specific services

### CODE REVIEW

**Comment by Pryz** [comment]
Should we make this optional to avoid blowing up the cardinality ?

**Comment by systemizer** [comment]
Yeah im up for that (mentioned that in the description). I can make it configurable

**Comment by systemizer** [comment]
@Pryz this is ready for review. no rush

File: httpstats/handler.go [hunk 1]
```diff
@@ -118,5 +140,12 @@ func (w *responseWriter) complete() {
 	}
 
 	w.metrics.observeResponse(res, "write", w.bytes, now.Sub(w.start))
+
+	// UserAgent can result in high cardinality. Allow mechanism for disabling the tag
+	if w.conf.DisableUserAgent {
+		w.metrics.http.userAgent = ""
+
```
Comment (Pryz) [inline httpstats/handler.go]: nit

Comment (systemizer) [reply httpstats/handler.go]: sorry, what's the nit you're referring to?

**Comment by f2prateek** [comment]
One thing I think could be helpful is we had a nicer API to augment the default stats emitted by this package, instead of manually adding options for things like path and userAgent. Something like this https://github.com/segmentio/stats/pull/100/files, that relies on the context for instance.

**Comment by f2prateek** [comment]
If we do go with this approach, I think for the sake of API consistency, it might be better instead to use the `DefaultFilters` option like we do for `http_path` - https://github.com/segmentio/stats/blob/17e5e763373e9d6a80804ff4dab85a842ad03a09/datadog/client.go#L34 (clients can disable it like this https://github.com/segmentio/ctlstore/pull/14/files).

**Comment by abraithwaite** [comment]
This is possible now with #111

**Pull request closed without merging**